frontend/src/App.tsx

```diff
@@ -598,30 +598,30 @@ export class App extends PureComponent<Props, State> {
 
   /**
    * Removes old elements. The term old is defined as:
-   *  - simple elements whose reportIds are no longer current
+   *  - element or container whose reportId is no longer current
+   *  - empty container
    */
   clearOldElements = (elements: any, reportId: string): BlockElement => {
     return elements
       .map((reportElement: ReportElement) => {
         const simpleElement = reportElement.get("element")
+        const isCurrent = reportElement.get("reportId") === reportId
```

Comment:
i think you can just shortcut out of this whole thing if this is not current reportId? Then you could just remove the isCurrent from each of the returns

```suggestion
        if (reportElement.get("reportId") !== reportId) return null
```

Reply:
Done!

(I thought about this briefly, and wasn't sure I wanted to risk the change in case they're not functionally equivalent. But I thought through it more, and your vote of confidence helps!)